lighthouse-core/gather/gatherers/same-origin-resources.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const url = require('url');
+const Gatherer = require('./gatherer');
+
+class SameOriginResources extends Gatherer {
```

Comment:
the `initialUrl` isn't passed in, but a gatherer could do that pretty easily (or URL could do both or something)